src/pages/Facility.tsx

```diff
@@ -13,8 +15,10 @@ export const Facility = () => {
 
   const facilityOffers = useMemo(
     () =>
-      facility !== undefined
-        ? offers.filter((offer) => offer.pricePlansReferences[facility.id] !== undefined)
+      facility !== undefined && offers !== undefined
+        ? (offers as (Offer & GenericStateRecord)[]).filter(
```

Comment:
I recommend defining OfferRecord instead of `Offer & GenericStateRecord`
and point your attention. If you required to add `as type` that is means that your Typescript code somewhere wrong.

Reply:
✔️

Reply:
```
export interface OfferRecord extends Offer {
  id: string;
}
```
now used instead `Offer & GenericStateRecord`

Reply:
![Screenshot 2022-08-05 at 16 26 21](https://user-images.githubusercontent.com/47505482/183097842-c6d18fbd-87e2-4142-ba99-22d850ff8d3d.jpg)
`as type` solves conflicts with `unknown` option of the record, `unknown` solves store conflicts

Reply:
`offers` are `unknown` of type because you commented it but why?

Reply:
GenericStateRecord is now merged with Facility and Offfer interfaces